### PR TITLE
Generalization to 3D

### PR DESCRIPTION
**Pull request by amboschman**
Selection of polytope type (lines 105-109) is now done via a conditional statement: is there a more elegant way to do this?

### CODE REVIEW

File: bulk_ghost_penalty_canvas.jl [hunk 1]
```diff
@@ -93,24 +95,18 @@ function setup_aggregates_bounding_box_model(bgmodel, aggregate_to_cells)
...
     end
 
-    # Set up the discrete model of bounding boxes 
-    ptr  = [ (((i-1)*2^D)+1) for i in 1:length(aggregate_to_cells)+1 ]
-    data = collect(1:length(bounding_box_node_coords))
-    bounding_box_node_ids = Gridap.Arrays.Table(data,ptr)
-
-    # TO-DO: this polytope is only valid for D=2
-    polytope=QUAD
+    # Set up the discrete model of bounding boxes
```
Comment (amartinhuertas) [inline bulk_ghost_penalty_canvas.jl:104]: Here you may use:

```
polytope = Polytope(Fill(HEX_AXIS,D)...)
```

to generalize the construction of the polytope to arbitrary space dimension. 

As we discussed, not a big deal ... we are going to run the code up to D=3 in practice.

File: bulk_ghost_penalty_canvas.jl [hunk 2]
```diff
@@ -93,24 +95,18 @@ function setup_aggregates_bounding_box_model(bgmodel, aggregate_to_cells)
...
-        #        Ideally, we should be able to generalize it for 
-        #        arbitrary D
-        for i in (xmin[2],xmax[2])
-            for j in (xmin[1],xmax[1])
-                bounding_box_node_coords[current]=Point(j,i)
-                current+=1
-            end 
-        end
+        bounds         = [(xmin[i], xmax[i]) for i in 1:D]
+        point_iterator = Iterators.product([bounds[i] for i in 1:D]...)
```
Comment (amartinhuertas) [inline bulk_ghost_penalty_canvas.jl:99]: can't you just use here `Iterators.product(bounds...)` ?

File: bulk_ghost_penalty_canvas.jl [hunk 3]
```diff
@@ -123,7 +119,7 @@ function setup_aggregates_bounding_box_model(bgmodel, aggregate_to_cells)
 end 
 
 aggregate_to_cells=setup_aggregate_to_cells(aggregates)
-aggregates_bounding_box_model=
+aggregates_bounding_box_model =
```
Comment (amartinhuertas) [inline bulk_ghost_penalty_canvas.jl:122]: Remove introduced blank space

**Review by amartinhuertas — CHANGES_REQUESTED** [review]
hi @amboschman  ... great work! Just minor comments for improvement (mainly so that we also get familiarized with the workflow).

File: bulk_ghost_penalty_canvas.jl (hunk 1 above)
Comment (amboschman) [reply bulk_ghost_penalty_canvas.jl:104]: Cheers, this is indeed what I was looking for!

File: bulk_ghost_penalty_canvas.jl [hunk 4]
```diff
@@ -93,24 +95,18 @@ function setup_aggregates_bounding_box_model(bgmodel, aggregate_to_cells)
...
-        for i in (xmin[2],xmax[2])
-            for j in (xmin[1],xmax[1])
-                bounding_box_node_coords[current]=Point(j,i)
-                current+=1
-            end 
-        end
+        bounds         = [(xmin[i], xmax[i]) for i in 1:D]
+        point_iterator = Iterators.product([bounds[i] for i in 1:D]...)
+        bounding_box_node_coords[bounding_box_node_ids[agg]] = 
+                reduce(vcat,[Point(p...) for p in point_iterator])
```
[the remark this answers is not in the document]
Comment (amboschman) [reply bulk_ghost_penalty_canvas.jl]: Makes sense, thanks!

**Review by amartinhuertas — APPROVED** [review]
Thanks!

**Pull request merged**